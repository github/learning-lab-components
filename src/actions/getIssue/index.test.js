const getIssue = require('./index')
const mockContext = require('../../tests/mockContext')

describe('getIssue', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      issue: {
        number: 1
      }
    }, {
      issues: {
        get: jest.fn(() => Promise.resolve({ data: { number: 1 } }))
      },
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My issue', number: 1 }]
          }
        }))
      }
    })
  })

  it('gets an issue', async () => {
    const issue = await getIssue(context, {})
    expect(issue).toMatchSnapshot()
  })

  it('gets an issue with a title string', async () => {
    const issue = await getIssue(context, { issue: 'My issue' })
    expect(issue).toMatchSnapshot()
  })

  it('gets an issue with a issue number', async () => {
    const issue = await getIssue(context, { issue: 1 })
    expect(issue).toMatchSnapshot()
  })
})
