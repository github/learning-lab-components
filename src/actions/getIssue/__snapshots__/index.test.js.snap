// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIssue gets an issue 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;

exports[`getIssue gets an issue with a issue number 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;

exports[`getIssue gets an issue with a title string 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;
